Northstar.Custom/mod/scripts/vscripts/sh_message_utils.gnut

```diff
@@ -0,0 +1,510 @@
+}
+
+void function SelectPollOption_Threaded( int index )
+{
+	if ( index >= client.poll.ruis.len() || index <= 0 )
+		return
+	
+	RuiSetFloat3( client.poll.ruis[ index ], "msgColor", ColorSelected )
+	EmitSoundOnEntity( GetLocalClientPlayer(), "menu_accept" )
+	
+	float endTime = 1 + client.poll.duration
+	while( endTime > Time() && client.poll.pollActive )
+		WaitFrame()
+	
+	GetLocalClientPlayer().ClientCommand( "poll_respond " + index )
+	
+	foreach( var rui in client.poll.ruis )
+		RuiDestroyIfAlive( rui )
+	
+	client.poll.ruis.clear()
+	client.poll.pollActive = false
+}
+
+void function ShowPollMessage_Threaded()
+{
+	if( client.poll.pollActive )
+		return
+	
+	client.poll.pollActive = true
+	
+	
+	for( int i = 0; i < client.poll.options.len() + 1; i++ )
+	{
+		var rui = CreateCockpitRui( $"ui/cockpit_console_text_top_left.rpak" )
+		// This makes it fade and me no likey >:(
+		//RuiSetInt( rui, "maxLines", 1 )
+		//RuiSetInt( rui, "lineNum", i )
+		RuiSetFloat2( rui, "msgPos", <0, 0.4 + i * 0.025, 0> )
+		if( i == 0 )
+		{
+			RuiSetFloat3( rui, "msgColor", ColorSelected )
+			RuiSetString( rui, "msgText", client.poll.header )
+		}
+		else
+		{
+			RuiSetFloat3( rui, "msgColor", ColorBase )
+			RuiSetString( rui, "msgText", i + ". " + client.poll.options[i - 1]  )
+		}
+		
+		RuiSetFloat( rui, "msgFontSize", 30.0 )
+		RuiSetFloat( rui, "msgAlpha", 0.9 )
+		RuiSetFloat( rui, "thicken", 0.0 )
+		
+		client.poll.ruis.append( rui )
+	}
+	
+	client.poll.options.clear()
+	
+	float endTime = Time() + client.poll.duration
+	while( endTime > Time() && client.poll.pollActive )
+		WaitFrame()
+	
+	
+	foreach( var rui in client.poll.ruis )
+		RuiDestroyIfAlive( rui )
+	
+	client.poll.ruis.clear()
+	client.poll.pollActive = false
+}
+
+void function InfoMessageHandler_Threaded()
+{
+	while( true )
+	{
+		while( client.infoMessageQueue.len() == 0 )
+			WaitFrame()
+
+		var rui = CreatePermanentCockpitRui( $"ui/death_hint_mp.rpak" )
+		RuiSetString( rui, "hintText", client.infoMessageQueue[0].description )
+		RuiSetGameTime( rui, "startTime", Time() )
+		RuiSetFloat3( rui, "bgColor", <0,0,0> )
+		RuiSetFloat( rui, "bgAlpha", 0.5 )
+
+		wait 7
+
+		client.infoMessageQueue.remove(0)
+		RuiDestroyIfAlive( rui )
+	}
+}
+
+void function AnnouncementMessageHandler_Threaded()
+{
+	while( true )
+	{
+		while( client.announcementQueue.len() == 0 )
+			WaitFrame()
+		
+		AnnouncementData announcement = Announcement_Create( client.announcementQueue[0].title )
+		Announcement_SetSubText( announcement, client.announcementQueue[0].description )
+		Announcement_SetTitleColor( announcement, client.announcementQueue[0].color )
+		Announcement_SetPurge( announcement, true )
+		Announcement_SetPriority( announcement, client.announcementQueue[0].priority )
+		Announcement_SetSoundAlias( announcement, SFX_HUD_ANNOUNCE_QUICK )
+		Announcement_SetStyle( announcement, client.announcementQueue[0].style )
+		AnnouncementFromClass( GetLocalViewPlayer(), announcement )
+
+		wait 5
```

Comment:
should make this a constant, convar or able to set within the command that calls this function. just waiting however long you think is nice might fuck you over in the future